From f1828ab5e78492a15c8e2ec9bb2fcd46a871bfea Mon Sep 17 00:00:00 2001
From: Biju Chakkachattil <bijux.chakkachattil@intel.com>
Date: Fri, 26 Jun 2015 03:32:38 +0530
Subject: [PATCH] EM/GPIO:-Use gpiod_set_value_cansleep function to avoid slow
 path warning

While connecting /Disconnecting USB/OTG,  slowpath dump stack is
printed due to  "can_sleep" flag is set to true. To avoid the
dump stack we need to use use "gpiod_set_value_cansleep" function
instead "gpiod_set_value".

Change-Id: I51babc0e8b4eac39cf9e01b06660812201f7f8d3
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-40198
Signed-off-by: Biju Chakkachattil <bijux.chakkachattil@intel.com>
---
 drivers/misc/intel_wcove_gpio.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/intel_wcove_gpio.c b/drivers/misc/intel_wcove_gpio.c
index aef5276a9726..a36370ff0dc3 100644
--- a/drivers/misc/intel_wcove_gpio.c
+++ b/drivers/misc/intel_wcove_gpio.c
@@ -78,12 +78,13 @@ static void wcove_gpio_ctrl_worker(struct work_struct *work)
 		mutex_lock(&info->lock);
 		/* set high only when otg connected */
 		if (!evt->is_sdp_connected)
-			gpiod_set_value(info->gpio_otg, evt->is_otg_connected);
+			gpiod_set_value_cansleep(info->gpio_otg,
+							evt->is_otg_connected);
 		/**
 		 * set high when sdp/otg connected and set low when sdp/otg
 		 * disconnected
 		 */
-		gpiod_set_value(info->gpio_vchgrin,
+		gpiod_set_value_cansleep(info->gpio_vchgrin,
 			(evt->is_sdp_connected ||
 				evt->is_otg_connected) ? 1 : 0);
 		mutex_unlock(&info->lock);
-- 
1.9.1

